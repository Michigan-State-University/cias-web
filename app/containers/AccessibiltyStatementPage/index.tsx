import React from 'react';
import Box from 'components/Box';

const AccessibilityStatementPage = () => (
  <Box padding={20}>
    <h1>Accessibility statement for CIAS</h1>

    <p>
      We want everyone who visits the CIAS website to feel welcome and find the
      experience rewarding.
    </p>

    <h2>What are we doing?</h2>

    <p>
      To help us make the CIAS website a positive place for everyone, we&apos;ve
      been using the{' '}
      <a href="http://www.w3.org/TR/WCAG/">
        Web Content Accessibility Guidelines (WCAG) 2.1
      </a>
      . These guidelines explain how to make web content more accessible for
      people with disabilities, and user friendly for everyone.
    </p>

    <p>
      The guidelines have three levels of accessibility (A, AA and AAA).
      We&apos;ve chosen Level AA as the target for the CIAS website.
    </p>

    <h2>How are we doing?</h2>

    <p>
      We&apos;ve worked hard on the CIAS website and believe we&apos;ve achieved
      our goal of Level AA accessibility. We monitor the website regularly to
      maintain this, but if you do find any problems, please get in touch.
    </p>

    <h2>Let us know what you think</h2>

    <p>
      If you enjoyed using the CIAS website, or if you had trouble with any part
      of it, please get in touch. We&apos;d like to hear from you in any of the
      following ways:
    </p>

    <ul>
      <li>
        email us at <a href="mailto:contact@cias.app">contact@cias.app</a>
      </li>

      <li>
        get in touch at{' '}
        <a href="https://msu.co1.qualtrics.com/jfe/form/SV_6S9MV7pPpoqFwTY">
          https://msu.co1.qualtrics.com/jfe/form/SV_6S9MV7pPpoqFwTY
        </a>
      </li>
    </ul>
  </Box>
);

export default AccessibilityStatementPage;
