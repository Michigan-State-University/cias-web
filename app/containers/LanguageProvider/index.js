/*
 *
 * LanguageProvider
 *
 * this component connects the redux state language locale to the
 * IntlProvider component and i18n messages (loaded from `app/translations`)
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import { IntlProvider } from 'react-intl';

import { themeColors } from 'theme';

import { makeSelectLocale } from './selectors';

// Those tags are handled by <Markup />
const errorsToOmit = ['INVALID_TAG', 'UNCLOSED_TAG'];

export const intlProviderConfig = {
  defaultRichTextElements: {
    space: (chunks) => (
      <>
        {'\u2000'}
        {chunks}
      </>
    ),
    bold: (chunks) => <b>{chunks}</b>,
    medium: (chunks) => <span style={{ fontWeight: 500 }}>{chunks}</span>,
    span: (chunks) => `<span>${chunks}</span>`,
    p: (chunks) => `<p>${chunks}</p>`,
    b: (chunks) => `<b>${chunks}</b>`,
    h1: (chunks) => `<h1>${chunks}</h1>`,
    h2: (chunks) => `<h2>${chunks}</h2>`,
    h3: (chunks) => `<h3>${chunks}</h3>`,
    ol: (chunks) => `<ol>${chunks}</ol>`,
    ul: (chunks) => `<ul>${chunks}</ul>`,
    li: (chunks) => `<li>${chunks}</li>`,
    sup: (chunks) => `<sup>${chunks}</sup>`,
    br: () => `<br />`,
    secondaryColorBold: (chunks) => (
      <span style={{ color: themeColors.secondary, fontWeight: 'bold' }}>
        {chunks}
      </span>
    ),
    primaryColorBold: (chunks) => (
      <span style={{ color: themeColors.primary, fontWeight: 'bold' }}>
        {chunks}
      </span>
    ),
    textColorBold: (chunks) => (
      <span style={{ color: themeColors.text, fontWeight: 'bold' }}>
        {chunks}
      </span>
    ),
    primaryColor: (chunks) => (
      <span style={{ color: themeColors.primary }}>{chunks}</span>
    ),
    warningColor: (chunks) => (
      <span style={{ color: themeColors.warning }}>{chunks}</span>
    ),
    phone: (chunks) => (
      <a
        href={`tel:${chunks}`}
        style={{ color: themeColors.primary, fontWeight: 500 }}
      >
        {chunks}
      </a>
    ),
  },
  onError: (error) => {
    const showError = !errorsToOmit.some((errorToOmit) =>
      error.toString().includes(errorToOmit),
    );
    // eslint-disable-next-line no-console
    if (showError) console.error(error);
  },
};

export function LanguageProvider(props) {
  return (
    <IntlProvider
      locale={props.locale}
      key={props.locale}
      messages={props.messages[props.locale]}
      {...intlProviderConfig}
    >
      {React.Children.only(props.children)}
    </IntlProvider>
  );
}

LanguageProvider.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object,
  children: PropTypes.element.isRequired,
};

const mapStateToProps = createSelector(makeSelectLocale(), (locale) => ({
  locale,
}));

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(LanguageProvider);
