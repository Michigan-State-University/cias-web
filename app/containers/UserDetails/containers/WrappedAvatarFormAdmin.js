import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import {
  makeSelectUser,
  addSingleUserAvatarSaga,
  addOtherUserAvatarRequest,
  deleteSingleUserAvatarSaga,
  deleteOtherUserAvatarRequest,
} from 'global/reducers/user';
import { useInjectSaga } from 'redux-injectors';

import AvatarForm from 'components/AccountSettings/AvatarForm';

const WrappedAvatarFormAdmin = ({
  userState: { user },
  addAvatar,
  deleteAvatar,
}) => {
  useInjectSaga({ key: 'addSingleUserAvatar', saga: addSingleUserAvatarSaga });
  useInjectSaga({ key: 'deleteAvatar', saga: deleteSingleUserAvatarSaga });

  const addAvatarCall = (avatarData) =>
    addAvatar({ userId: user.id, ...avatarData });

  const deleteAvatarCall = () => deleteAvatar(user.id);
  return (
    <AvatarForm
      user={user}
      addAvatar={addAvatarCall}
      deleteAvatar={deleteAvatarCall}
    />
  );
};

WrappedAvatarFormAdmin.propTypes = {
  userState: PropTypes.object,
  addAvatar: PropTypes.func,
  deleteAvatar: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  userState: makeSelectUser(),
});
const mapDispatchToProps = {
  addAvatar: addOtherUserAvatarRequest,
  deleteAvatar: deleteOtherUserAvatarRequest,
};

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(withConnect)(WrappedAvatarFormAdmin);
