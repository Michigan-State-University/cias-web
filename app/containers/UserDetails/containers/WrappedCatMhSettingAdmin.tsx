import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useIntl } from 'react-intl';
import { useInjectSaga } from 'redux-injectors';

import {
  makeSelectUser,
  editOtherUserRequest,
  editSingleUserSaga,
} from 'global/reducers/user';

import Switch, { LabelPosition } from 'components/Switch';
import Box from 'components/Box';
import Text from 'components/Text';

import messages from '../messages';

const WrappedCatMhSetting = (): JSX.Element | null => {
  useInjectSaga({ key: 'editSingleUser', saga: editSingleUserSaga });

  const { formatMessage } = useIntl();
  const dispatch = useDispatch();
  const { user } = useSelector(makeSelectUser());

  if (!user) {
    return null;
  }

  const { abilityToCreateCatMh, id: userId } = user;

  const editUserCall = () =>
    dispatch(
      editOtherUserRequest({
        userId,
        abilityToCreateCatMh: !abilityToCreateCatMh,
      }),
    );

  return (
    <Box display="flex" align="center">
      <Switch
        disabled={false}
        id="ability-to-create-cat-mh"
        checked={abilityToCreateCatMh}
        onToggle={editUserCall}
        labelPosition={LabelPosition.Right}
      >
        <Text ml={10}>{formatMessage(messages.canCreateCatMhSession)}</Text>
      </Switch>
    </Box>
  );
};

export default WrappedCatMhSetting;
