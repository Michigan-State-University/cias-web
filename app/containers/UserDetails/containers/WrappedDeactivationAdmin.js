import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import {
  makeSelectUser,
  changeActivateStatusRequest,
  changeActivateStatusSaga,
} from 'global/reducers/user';
import { useInjectSaga } from 'utils/injectSaga';

import Deactivation from 'components/AccountSettings/Deactivation';

const WrappedDeactivationAdmin = ({ userState: { user }, changeStatus }) => {
  useInjectSaga({ key: 'changeActiveStatus', saga: changeActivateStatusSaga });

  const changeStatusCall = newStatus => changeStatus(user.id, newStatus);

  return <Deactivation user={user} changeStatus={changeStatusCall} />;
};

WrappedDeactivationAdmin.propTypes = {
  userState: PropTypes.object,
  changeStatus: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  userState: makeSelectUser(),
});
const mapDispatchToProps = {
  changeStatus: changeActivateStatusRequest,
};

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(WrappedDeactivationAdmin);
