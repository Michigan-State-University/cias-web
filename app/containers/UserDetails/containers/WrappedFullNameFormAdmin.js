import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import {
  makeSelectUser,
  editOtherUserRequest,
  editSingleUserSaga,
} from 'global/reducers/user';
import { useInjectSaga } from 'redux-injectors';

import FullNameForm from 'components/AccountSettings/FullNameForm';

const WrappedFullNameForm = ({ formatMessage, user, editUser }) => {
  useInjectSaga({ key: 'editSingleUser', saga: editSingleUserSaga });
  const editUserCall = (userData) => editUser({ userId: user.id, ...userData });
  if (!user) {
    return null;
  }
  return (
    <FullNameForm
      formatMessage={formatMessage}
      user={user}
      editUser={editUserCall}
    />
  );
};

WrappedFullNameForm.propTypes = {
  formatMessage: PropTypes.func,
  user: PropTypes.object,
  editUser: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  user: makeSelectUser(),
});

const mapDispatchToProps = {
  editUser: editOtherUserRequest,
};

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(withConnect, injectIntl)(WrappedFullNameForm);
