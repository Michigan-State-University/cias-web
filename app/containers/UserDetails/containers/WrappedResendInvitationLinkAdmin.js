import React, { useCallback, useMemo } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useInjectSaga } from 'redux-injectors';

import {
  makeSelectUser,
  makeSelectUserLoader,
  resendInvitationLinkRequest,
  resendInvitationLinkSaga,
} from 'global/reducers/user';

import { ResendInvitationLink } from 'components/AccountSettings/ResendInvitationLink';

export const WrappedResendInvitationLinkAdmin = () => {
  const dispatch = useDispatch();
  useInjectSaga({ key: 'user', saga: resendInvitationLinkSaga });

  const user = useSelector(makeSelectUser());
  const loading = useSelector(makeSelectUserLoader('resendInvitationLink'));

  const resendInvitation = useCallback(() => {
    dispatch(resendInvitationLinkRequest(user?.id));
  }, [user?.id]);

  const activated = useMemo(() => user?.fullName?.trim(), [user?.fullName]);

  if (!user || activated) return <></>;

  return <ResendInvitationLink loading={loading} onResend={resendInvitation} />;
};
