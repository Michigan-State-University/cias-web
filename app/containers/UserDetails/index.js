/**
 *
 * UserDetails
 *
 */

import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import {
  makeSelectUser,
  fetchUserRequest,
  fetchUserSaga,
  UserReducer,
  makeSelectUserLoader,
  makeSelectUserError,
} from 'global/reducers/user';

import { injectReducer, injectSaga } from 'redux-injectors';

import Spinner from 'components/Spinner';
import { themeColors } from 'theme';
import ErrorAlert from 'components/ErrorAlert';
import AccountSettings from '../AccountSettings';
import WrappedAvatarFormAdmin from './containers/WrappedAvatarFormAdmin';
import WrappedFullNameFormAdmin from './containers/WrappedFullNameFormAdmin';
import WrappedDeactivationAdmin from './containers/WrappedDeactivationAdmin';
import { WrappedResendInvitationLinkAdmin } from './containers/WrappedResendInvitationLinkAdmin';

export const UserDetails = ({
  user,
  userError,
  userLoading,
  fetchUser,
  match: {
    params: { id },
  },
}) => {
  useEffect(() => {
    fetchUser(id);
  }, []);

  if (userLoading) {
    return <Spinner color={themeColors.secondary} />;
  }
  if (userError) {
    return <ErrorAlert errorText={userError} />;
  }

  return (
    <AccountSettings
      NotificationsComponent={null}
      formComponents={{
        TimezoneComponent: null,
        EmailComponent: null,
        PasswordComponent: null,
        AvatarComponent: WrappedAvatarFormAdmin,
        FullNameComponent: WrappedFullNameFormAdmin,
        DeactivationComponent: WrappedDeactivationAdmin,
        ResendInvitationLinkComponent: WrappedResendInvitationLinkAdmin,
      }}
      userId={user.id}
    />
  );
};

UserDetails.propTypes = {
  user: PropTypes.object,
  userLoading: PropTypes.bool,
  userError: PropTypes.object,
  match: PropTypes.object,
  fetchUser: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  user: makeSelectUser(),
  userLoading: makeSelectUserLoader('user'),
  userError: makeSelectUserError('user'),
});

const mapDispatchToProps = {
  fetchUser: fetchUserRequest,
};

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  injectSaga({ key: 'userSaga', saga: fetchUserSaga }),
  injectReducer({ key: 'user', reducer: UserReducer }),
)(UserDetails);
