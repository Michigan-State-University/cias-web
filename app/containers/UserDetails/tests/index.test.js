/**
 *
 * Tests for UserDetails
 *
 */

import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { DEFAULT_LOCALE } from 'i18n';

import { createTestStore } from 'utils/testUtils/storeUtils';
import { Roles } from 'models/User/UserRoles';
import { initialState as authState } from 'global/reducers/auth';
import { intlProviderConfig } from 'containers/LanguageProvider';
import { UserDetails } from '../index';

describe('<UserDetails />', () => {
  let props;
  const initialState = {
    auth: {
      ...authState,
      user: {
        roles: [Roles.admin],
      },
    },
    user: {
      user: {
        id: '1',
        full_name: 'test test',
        roles: ['participant'],
        email: 'test@test.pl',
        avatar: null,
        firstName: 'test',
        lastName: 'test',
        active: true,
      },
      loaders: {
        user: false,
        resendInvitationLink: false,
      },
      errors: {
        user: null,
        resendInvitationLink: null,
      },
    },
  };
  let store;
  let modalContainer;
  let mainAppContainer;
  beforeEach(() => {
    store = createTestStore(initialState);

    ReactDOM.createPortal = jest.fn((element) => element);
    modalContainer = document.createElement('div');
    modalContainer.setAttribute('id', 'modal-portal');
    document.body.appendChild(modalContainer);
    mainAppContainer = document.createElement('div');
    mainAppContainer.setAttribute('id', 'main-app-container');
    document.body.appendChild(mainAppContainer);

    props = {
      user: {
        id: '1',
        full_name: 'test test',
        roles: ['participant'],
        email: 'test@test.pl',
        avatar: null,
        active: true,
      },
      match: {
        params: {
          id: '1',
        },
      },
      fetchUser: jest.fn(),
    };
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE} {...intlProviderConfig}>
          <MemoryRouter>
            <UserDetails {...props} />
          </MemoryRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE} {...intlProviderConfig}>
          <MemoryRouter>
            <UserDetails {...props} />
          </MemoryRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
