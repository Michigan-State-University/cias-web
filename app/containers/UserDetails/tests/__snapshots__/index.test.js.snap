// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetails /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="Box-sc-1vhutan-0 styled__StyledBox-sc-18ethpf-0 dISrsd jmOGcK"
    height="100%"
    width="100%"
  >
    <a
      class="styled__StyledLink-sc-3pirdo-0 jceFfF"
      href="/users"
    >
      <img
        alt="traingle"
        class="Img-sc-55j3su-0 gfzPuQ"
        src="IMAGE_MOCK"
      />
      <p
        class="Text-e4p0w3-0 cFbubw"
        color="#B727EA"
        font-weight="bold"
      >
        Back to user list
      </p>
    </a>
    <h1
      class="H1-sc-1k7m23t-0 eMTspO"
    >
      Account Settings
    </h1>
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 100%; flex-grow: 0; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <div
          class="Box-sc-1vhutan-0 iCDuTP"
        >
          <h2
            class="H2-uw8lbl-0 iUhbVg"
          >
            Profile
          </h2>
          <div
            class="Row-y1cs0u-0 cUkLOH"
            display="flex"
          >
            <div>
              <div
                class="Row-y1cs0u-0 fgwsqb"
                width="100%"
              >
                <div
                  class="styled__AvatarStyled-bn66gh-0 fvqhMv styled__StyledUserAvatar-sc-4bx7ap-1 dObBuO"
                  color="#FFFFFF"
                >
                  <div>
                    TT
                  </div>
                </div>
                <div
                  class="Row-y1cs0u-0 jOTCVU"
                >
                  <div
                    class="Row-y1cs0u-0 styled__StyledButtonsRow-sc-4bx7ap-2 dskFsL fA-DhtD"
                  >
                    <div
                      class="Box-sc-1vhutan-0 crjFXF styled__StyledUploadFileButton-sc-4bx7ap-4 GlRFp"
                      height="fit-content"
                    >
                      <input
                        autocomplete="off"
                        style="display: none;"
                        tabindex="-1"
                        type="file"
                      />
                      <div
                        class="Row-y1cs0u-0 bLAOxT"
                      >
                        <p
                          class="Text-e4p0w3-0 gEakrP"
                          color="#B727EA"
                          font-size="14"
                          font-weight="bold"
                        >
                          Add profile photo
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <button
                class="StyledButton-d2s0gt-0 eHTczx"
                color="warning"
                radius="100px"
                width="180"
              >
                Deactivate account
              </button>
            </div>
          </div>
          <div
            class="Row-y1cs0u-0 eDtOGP"
          >
            <div
              class="Row-y1cs0u-0 styled__StyledFullNameRow-sc-4bx7ap-0 gQFDhs loGlNM"
              width="100%"
            >
              <div
                class="Column-sc-4zswzq-0 ehOEzE"
                data-testid="first-name-input"
              >
                <label
                  for="firstName"
                >
                  <p
                    class="Text-e4p0w3-0 kFFPgx"
                    width="fit-content"
                  >
                    First Name
                  </p>
                </label>
                <div
                  class="Row-y1cs0u-0 fgwsqb"
                  width="100%"
                >
                  <input
                    aria-label="Enter first name"
                    class="Input-t9j2q5-0 ptbLy"
                    height="46"
                    id="firstName"
                    name="firstName"
                    placeholder="Enter first name"
                    type="text"
                    value="test"
                    width="100%"
                  />
                </div>
              </div>
              <div
                class="Column-sc-4zswzq-0 jCuscc"
              >
                <label
                  for="lastName"
                >
                  <p
                    class="Text-e4p0w3-0 kFFPgx"
                    width="fit-content"
                  >
                    Last name
                  </p>
                </label>
                <div
                  class="Row-y1cs0u-0 fgwsqb"
                  width="100%"
                >
                  <input
                    aria-label="Enter last name"
                    class="Input-t9j2q5-0 ptbLy"
                    height="46"
                    id="lastName"
                    name="lastName"
                    placeholder="Enter last name"
                    type="text"
                    value="test"
                    width="100%"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="Row-y1cs0u-0 styled__StyledRow-sc-1pif2s-1 fyfnXk jKoKLe"
            width="100%"
          />
          <div
            class="Row-y1cs0u-0 styled__StyledRow-sc-1pif2s-1 fyfnXk jKoKLe"
            width="100%"
          >
            <div
              class="Column-sc-4zswzq-0 ekpwZH"
            >
              <div
                class="Row-y1cs0u-0 hqAUhv"
                width="100%"
              >
                <div
                  class="Column-sc-4zswzq-0 HRVgp"
                  width="230"
                >
                  <label
                    for="iso"
                  >
                    <p
                      class="Text-e4p0w3-0 kFFPgx"
                      width="fit-content"
                    >
                      Phone prefix
                    </p>
                  </label>
                  <div
                    class="Box-sc-1vhutan-0 dvfkZg"
                    data-testid="select"
                  >
                    <div
                      class=" css-14jk2my-container"
                    >
                      <span
                        aria-atomic="false"
                        aria-live="polite"
                        aria-relevant="additions text"
                        class="css-1f43avz-a11yText-A11yText"
                      />
                      <div
                        class=" css-vz7ypb-control"
                      >
                        <div
                          class=" css-g1d714-ValueContainer"
                        >
                          <div
                            class=" css-q5ew0g-placeholder"
                          >
                            Code
                          </div>
                          <div
                            class="css-1bwsx9e-Input"
                          >
                            <div
                              class=""
                              style="display: inline-block;"
                            >
                              <input
                                aria-autocomplete="list"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                disabled=""
                                id="iso"
                                spellcheck="false"
                                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                              <div
                                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class=" css-1hb7zxy-IndicatorsContainer"
                        >
                          <div
                            aria-hidden="true"
                            class=" css-tlfecz-indicatorContainer"
                          >
                            <img
                              alt="arrow"
                              class="Img-sc-55j3su-0 gTKGj"
                              src="IMAGE_MOCK"
                              transform=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="Column-sc-4zswzq-0 ekpwZH"
                  value=""
                >
                  <label
                    for="number"
                  >
                    <p
                      class="Text-e4p0w3-0 kFFPgx"
                      width="fit-content"
                    >
                      Phone number
                    </p>
                  </label>
                  <input
                    aria-label="Enter phone number"
                    class="Input-t9j2q5-0 dbKITh"
                    disabled=""
                    id="number"
                    name="number"
                    placeholder="Enter phone number"
                    type="tel"
                    value=""
                    width="100%"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="Row-y1cs0u-0 styled__StyledRow-sc-1pif2s-1 eomgnk jKoKLe"
            width="100%"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
